tests/end2end/test_invocations.py

```diff
@@ -177,3 +183,83 @@ def test_optimize(request, quteproc_new, capfd, level):
+    proc = QuteProc(request)
+    proc.proc.setProcessChannelMode(QProcess.SeparateChannels)
+
+    try:
+        proc.start(args)
+        proc.wait_for_quit()
+    except ProcessExited:
+        assert proc.proc.exitStatus() == QProcess.NormalExit
+        pass
+    else:
+        assert False
+
+    proc.proc.setReadChannel(QProcess.StandardOutput)
+    lines = []
+    while proc.proc.canReadLine():
+        line = proc.proc.readLine()
+        lines.append(bytes(line).decode('utf-8').rstrip('\r\n'))
+    output = '\n'.join(lines)
+
+    assert re.search('^qutebrowser\s+v\d+(\.\d+)*$',
+        output, re.MULTILINE) is not None
+    assert re.search('^CPython:\s+\d+(\.\d+)*$',
+        output, re.MULTILINE) is not None
+    assert re.search('^Qt:\s+\d+(\.\d+)*$',
+        output, re.MULTILINE) is not None
+    assert re.search('^PyQt:\s+\d+(\.\d+)*$',
+        output, re.MULTILINE) is not None
+    assert re.search('^sip:\s+(\d+(\.\d+)*|no)$',
+        output, re.MULTILINE) is not None
+    assert re.search('^colorama:\s+(\d+(\.\d+)*|no)$',
+        output, re.MULTILINE) is not None
+    assert re.search('^pypeg2:\s+(\d+(\.\d+)*|no)$',
+        output, re.MULTILINE) is not None
+    assert re.search('^jinja2:\s+(\d+(\.\d+)*|no)$',
+        output, re.MULTILINE) is not None
+    assert re.search('^pygments:\s+(\d+(\.\d+)*|no)$',
+        output, re.MULTILINE) is not None
+    assert re.search('^yaml:\s+(\d+(\.\d+)*|no)$',
+        output, re.MULTILINE) is not None
+    assert re.search('^cssutils:\s+(\d+(\.\d+)*.*|no)$',
+        output, re.MULTILINE) is not None
+    assert re.search('^typing:\s+(yes|no)$',
+        output, re.MULTILINE) is not None
+    assert re.search('^PyQt5\.QtWebEngineWidgets:\s+(yes|no)$',
+        output, re.MULTILINE) is not None
+    assert re.search('^pdf\.js:\s+(\d+(\.\d+)*|no)$',
+        output, re.MULTILINE) is not None
+    assert re.search('^Webkit:\s+(\d+(\.\d+)*|no)$',
+        output, re.MULTILINE) is not None
+    assert re.search('^SSL:\s.+$',
+        output, re.MULTILINE) is not None
+    assert re.search('^Platform:\s.+,\s+.+$',
+        output, re.MULTILINE) is not None
+    assert re.search('^Frozen:\s(True|False)$',
+        output, re.MULTILINE) is not None
+    assert re.search('^Imported from .+$',
+        output, re.MULTILINE) is not None
+    assert re.search('^OS Version: .*$',
+        output, re.MULTILINE) is not None
+    assert re.search('^Paths$',
+        output, re.MULTILINE) is not None
+    assert re.search('^config:\s+.+$',
+        output, re.MULTILINE) is not None
+    assert re.search('^system_data:\s+.+$',
+        output, re.MULTILINE) is not None
+    assert re.search('^cache:\s+.+$',
+        output, re.MULTILINE) is not None
+    assert re.search('^data:\s+.+$',
+        output, re.MULTILINE) is not None
+    assert re.search('^download:\s+.+$',
+        output, re.MULTILINE) is not None
+    assert re.search('^runtime:\s+.+$',
+        output, re.MULTILINE) is not None
```

Comment:
All this seems needlessly specific - I think checking for `qutebrowser v...` or so would be enough?